src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectGCDumpActionTests.cs

```diff
@@ -0,0 +1,127 @@
+
+                await TestHostHelper.CreateCollectionRulesHost(_outputHelper, rootOptions =>
+                {
+                    rootOptions.AddFileSystemEgress(ExpectedEgressProvider, uniqueEgressDirectory.FullName);
+
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .AddCollectGCDumpAction(ExpectedEgressProvider)
+                        .SetStartupTrigger();
+                }, async host =>
+                {
+                    IOptionsMonitor<CollectionRuleOptions> ruleOptionsMonitor = host.Services.GetService<IOptionsMonitor<CollectionRuleOptions>>();
+                    CollectGCDumpOptions options = (CollectGCDumpOptions)ruleOptionsMonitor.Get(DefaultRuleName).Actions[0].Settings;
+
+                    ICollectionRuleActionProxy action;
+                    Assert.True(host.Services.GetService<ICollectionRuleActionOperations>().TryCreateAction(KnownCollectionRuleActions.CollectGCDump, out action));
+
+                    EndpointInfoSourceCallback callback = new(_outputHelper);
+                    await using var source = _endpointUtilities.CreateServerSource(out string transportName, callback);
+                    source.Start();
+
+                    AppRunner runner = _endpointUtilities.CreateAppRunner(transportName, TargetFrameworkMoniker.Net60); // Arbitrarily chose Net60
+
+                    Task<IEndpointInfo> newEndpointInfoTask = callback.WaitForNewEndpointInfoAsync(runner, CommonTestTimeouts.StartProcess);
+
+                    await runner.ExecuteAsync(async () =>
+                    {
+                        IEndpointInfo endpointInfo = await newEndpointInfoTask;
+
+                        using CancellationTokenSource cancellationTokenSource = new CancellationTokenSource(CommonTestTimeouts.GCDumpTimeout);
+                        CollectionRuleActionResult result = await action.ExecuteAsync(options, endpointInfo, cancellationTokenSource.Token);
+
+                        Assert.NotNull(result.OutputValues);
+                        Assert.True(result.OutputValues.TryGetValue(CollectionRuleActionConstants.EgressPathOutputValueName, out string egressPath));
+                        Assert.True(File.Exists(egressPath));
+
+                        using FileStream gcdumpStream = new(egressPath, FileMode.Open, FileAccess.Read);
+                        Assert.NotNull(gcdumpStream);
+
+                        await ValidateGCDump(gcdumpStream);
+
+                        await runner.SendCommandAsync(TestAppScenarios.AsyncWait.Commands.Continue);
+                    });
+                });
+            }
+            finally
+            {
+                try
+                {
+                    uniqueEgressDirectory?.Delete(recursive: true);
+                }
+                catch
+                {
+                }
+            }
+        }
+
+        private static async Task ValidateGCDump(Stream gcdumpStream)
```

Comment:
This is loosely based off of the validation we currently do on Dumps. I don't know much about the composition of GCDumps, so this might be a poor approach, but I ended up cracking them open to look at the bytes instead of just checking that a file existed without examining its contents.